source/material_model/rheology/friction_options.cc

```diff
@@ -0,0 +1,200 @@
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/material_model/rheology/friction_options.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/base/quadrature_lib.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      double
+      FrictionOptions<dim>::
+      compute_dependent_friction_angle(const double current_edot_ii,
+                                       const unsigned int j,  // j is from a for-loop over volume_fractions.size()
+                                       double current_friction) const
+      {
+
+        switch (friction_dependence_mechanism)
+          {
+            case independent:
+            {
+              break;
+            }
+            case dynamic_friction:
+            {
+              // Calculate effective steady-state friction coefficient.
+              // This is based on the former material model dynamic friction.
+              // The formula below is equivalent to the equation 13 in \cite{van_dinther_seismic_2013}.
+              // Although here the dynamic friction coefficient is directly specified. In addition,
+              // we also use a reference strain rate in place of a characteristic
+              // velocity divided by local element size. This reference strain rate is called
+              // the dynamic characteristic strain rate and is used to see what value between
+              // dynamic and static angle of internal friction should be used.
+              // Furthermore a smoothness exponent X is added, which determines whether the
+              // friction vs strain rate curve is rather step-like or more gradual.
+              // mu  = mu_d + (mu_s - mu_d) / ( (1 + strain_rate_dev_inv2/dynamic_characteristic_strain_rate)^X );
+              // Angles of friction are used in radians within ASPECT. The coefficient
+              // of friction (mu) is the tangent of the internal angle of friction, hence convergence is needed.
+              const double mu = (std::tan(dynamic_angles_of_internal_friction[j])
+                                 + (std::tan(current_friction) - std::tan(dynamic_angles_of_internal_friction[j]))
```

Comment:
Why is current_friction used to compute current_friction? Because it holds the static_friction_angles right? Please explain that in a comment. Also, instead of current_friction, could you call it current_friction_angles in this function? Just to make the distinction between friction angle and friction coefficient clearer.